pytest_catchlog.py

```diff
@@ -92,19 +97,81 @@ def pytest_addoption(parser):
                     help='disable printing caught logs on failed tests.')
     add_option_ini(parser,
                    '--log-format',
-                   dest='log_format', default=('%(filename)-25s %(lineno)4d'
-                                               ' %(levelname)-8s %(message)s'),
+                   dest='log_format',
+                   default=DEFAULT_LOG_FORMAT,
                    help='log format as used by the logging module.')
     add_option_ini(parser,
                    '--log-date-format',
-                   dest='log_date_format', default=None,
+                   dest='log_date_format',
+                   default=DEFAULT_DATE_FORMAT,
                    help='log date format as used by the logging module.')
+    add_option_ini(parser,
+                   '--log-extra-level',
+                   dest='log_extra_levels',
+                   type='args',
+                   action='append',
+                   help='extra logging levels that verbosity should be sensible to.')
 
 
 def pytest_configure(config):
     """Always register the log catcher plugin with py.test or tests can't
     find the  fixture function.
     """
+    # Prepare the available levels dictionary
+    available_levels = set([
+        logging.CRITICAL,
+        logging.ERROR,
+        logging.WARNING,
+        logging.INFO,
+        logging.DEBUG,
+        logging.NOTSET
+    ])
+    for level in get_option_ini(config, 'log_extra_levels'):
+        try:
+            level_num = int(getattr(logging, level, level))
+        except ValueError:
+            # Python logging does not recognise this as a logging level
+            raise pytest.UsageError(
+                "'{0}' is not recognized as a logging level name. Please "
+                "consider passing the logging level num instead.".format(
+                    level))
+        else:
+            if not (logging.NOTSET <= level_num <= logging.CRITICAL):
+                raise pytest.UsageError(
+                    "'{0}' is ignored as not being in the valid logging "
+                    "levels range: NOTSET({1}) - CRITICAL({2})".format(
+                        level,
+                        logging.NOTSET,
+                        logging.CRITICAL))
+                continue
```

Comment:
Don't follow the excess part, its still a usage error  and your `config.warn` suggestion seemed too shy and easy to miss. You'd need to pass `-rw` to see it right?

And, we should stick with the same error reporting for both situations. So either the usage error, or the warn. Unless I'm missing something, which is also possible.

So, which one?